lib/Doctrine/Migrations/Configuration/ArrayConfiguration.php

```diff
@@ -4,6 +4,9 @@
 
 namespace Doctrine\Migrations\Configuration;
 
+/**
+ * @internal
```

Comment:
@jwage should these configuration classes really be internal ? How are they used then ?

Reply:
They don't get used directly, they are a part of the internals of the console commands but are never used directly. They produce a `Configuration` instance that then gets used in the console commands.

Reply:
should these classes actually be extending Configuration then ?

Reply:
@stof arguably no, they should not. A better design would be to inject the Configuration in to ArrayConfiguration, etc. and then call the public getters on Configuration and have no inheritance at all. I could try tackling that refactoring in another PR.